### PR TITLE
Update to Swift 3 (Xcode 8.3.1)

### PR DESCRIPTION
**Pull request by azubala**
**Notice** Please review only [this commit](https://github.com/mobile-academy/tdd-workshop-swift/pull/2/commits/7721f448277798215bab607c06a94528786a6183)

What's up:
- This PR makes the project to compile (both app and specs)
- Stubbed or commented out code which was not compiling or didn’t make sense afte Parse removal (marked with `TODO`s)
- Updated pods to the most current ones (using Cocoapods 1.2.1)
- Removed `Parse`
- Performed automatic migration from Swift 2 to Swift 3

---

IMHO this should be a starting point for each of us to make sure that _their_ corresponding part is functional.

### CODE REVIEW

File: TDDWorkshop/Classes/PhotoStream/Model/StreamItemCreatorSpec.swift [hunk 1]
```diff
@@ -83,23 +83,23 @@ class StreamItemCreatorSpec: QuickSpec {
...
+                        expect(alertAction.style) == UIAlertActionStyle.default
                     }
                     it("should present alert controller with 3rd option 'Cancel'") {
                         let alertAction = alertActionFactory.capturedActions[2]
                         expect(alertAction.title) == "Cancel"
                     }
                     it("should present alert controller with 3rd option set to Cancel style") {
                         let alertAction = alertActionFactory.capturedActions[2]
-                        expect(alertAction.style) == UIAlertActionStyle.Cancel
+                        expect(alertAction.style) == UIAlertActionStyle.cancel
```
Comment (literator) [inline TDDWorkshop/Classes/PhotoStream/Model/StreamItemCreatorSpec.swift:102]: I wonder if that compiles without `UIAlertActionStyle`, only `.cancel`?

File: TDDWorkshop/Classes/PhotoStream/PhotoStreamViewController.swift [hunk 2]
```diff
@@ -92,26 +96,26 @@ class PhotoStreamViewController: UICollectionViewController, ItemCreatingDelegat
...
-    private func setupCollectionView() {
-        refreshControl.addTarget(self, action: #selector(didPullToRefresh), forControlEvents: UIControlEvents.ValueChanged)
+    fileprivate func setupCollectionView() {
+        refreshControl.addTarget(self, action: #selector(didPullToRefresh), for: UIControlEvents.valueChanged)
         collectionView?.addSubview(refreshControl)
         collectionView?.alwaysBounceVertical = true
     }
 
-    private func downloadStreamItems() {
+    fileprivate func downloadStreamItems() {
```
Comment (literator) [inline TDDWorkshop/Classes/PhotoStream/PhotoStreamViewController.swift:118]: Depending on context, but I guess those could stay `private` (not sure now if those methods are used in extensions).

File: TDDWorkshop/Classes/PhotoStream/PhotoStreamViewControllerSpec.swift [hunk 3]
```diff
@@ -44,7 +44,7 @@ class PhotoStreamViewControllerSpec: QuickSpec {
                     var collectionViewFake : UICollectionViewFake!
 
                     beforeEach {
-                        collectionViewFake = UICollectionViewFake(frame: CGRectZero,
+                        collectionViewFake = UICollectionViewFake(frame: CGRect(),
```
Comment (literator) [inline TDDWorkshop/Classes/PhotoStream/PhotoStreamViewControllerSpec.swift:47]: `CGRect()` -> `.zero`

File: TDDWorkshop/Classes/PhotoStream/PhotoStreamViewControllerSpec.swift [hunk 4]
```diff
@@ -119,7 +119,7 @@ class PhotoStreamViewControllerSpec: QuickSpec {
                         var collectionViewFake : UICollectionViewFake!
 
                         beforeEach {
-                            collectionViewFake = UICollectionViewFake(frame: CGRectZero,
+                            collectionViewFake = UICollectionViewFake(frame: CGRect(),
```
Comment (literator) [inline TDDWorkshop/Classes/PhotoStream/PhotoStreamViewControllerSpec.swift:122]: `CGRect()` -> `.zero`

File: TDDWorkshop/Classes/PhotoStream/StreamItemViewController.swift [hunk 5]
```diff
@@ -31,7 +31,7 @@ class StreamItemViewController: UIViewController {
 
     //MARK: Actions
     
-    @IBAction func cancelBarButtonPressed(sender: AnyObject) {
-        self.dismissViewControllerAnimated(true) {}
+    @IBAction func cancelBarButtonPressed(_ sender: AnyObject) {
+        self.dismiss(animated: true) {}
```
Comment (literator) [inline TDDWorkshop/Classes/PhotoStream/StreamItemViewController.swift:35]: Personally, I'd write it `dismiss(animated: true, completion: nil)`. Empty closure looks weird, don't you think?

File: TDDWorkshop/Classes/Utils/MobileAcademyStyleSheet.swift [hunk 6]
```diff
@@ -24,12 +24,12 @@ extension UIColor {
 class MobileAcademyStyleSheet {
     static func applyStyle() {
         UINavigationBar.appearance().barTintColor = UIColor.ma_tealColor()
-        UINavigationBar.appearance().translucent = false
-        UINavigationBar.appearance().titleTextAttributes = [NSForegroundColorAttributeName : UIColor.whiteColor()]
-        UINavigationBar.appearance().tintColor = UIColor.whiteColor()
+        UINavigationBar.appearance().isTranslucent = false
+        UINavigationBar.appearance().titleTextAttributes = [NSForegroundColorAttributeName : UIColor.white]
+        UINavigationBar.appearance().tintColor = UIColor.white
```
Comment (literator) [inline TDDWorkshop/Classes/Utils/MobileAcademyStyleSheet.swift:29]: We can drop `UIColor` here and above, if type can be inferred.

**Review by literator — APPROVED** [review]
It's okay.

File: TDDWorkshop/Classes/PhotoStream/Model/StreamItemCreatorSpec.swift (hunk 1 above)
Comment (azubala) [reply TDDWorkshop/Classes/PhotoStream/Model/StreamItemCreatorSpec.swift:102]: Nope. It complains without it.

File: TDDWorkshop/Classes/PhotoStream/PhotoStreamViewController.swift (hunk 2 above)
Comment (azubala) [reply TDDWorkshop/Classes/PhotoStream/PhotoStreamViewController.swift:118]: This is automatic Swift 3 migration :) I would take a closer look when updating PhotoStream part.

File: TDDWorkshop/Classes/PhotoStream/StreamItemViewController.swift (hunk 5 above)
Comment (azubala) [reply TDDWorkshop/Classes/PhotoStream/StreamItemViewController.swift:35]: This is automatic Swift 3 migration :) Plus the previous version had `{}` :)

**Pull request merged**